### PR TITLE
Fix the updated version of perfscale load generator, VMIs are not being created

### LINKED ISSUE #7083: Prometheus rest_client_requests_total metric is missing pod DELETE and CREATE events
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
We identified this issue occurring in our performance test on the CI/CD system. The test deploys kubevirt, creates 100 VMIs, and cleans up the environment. Also, the test runs the [perfscale-audit tool](https://github.com/kubevirt/kubevirt/tree/main/tools/perfscale-audit) to collect some Prometheus metrics and dump the results into the test.

However, the results are very unstable, sometimes the DELETE and CREATE pod event doesn't even appear in the results.

**How to reproduce it (as minimally and precisely as possible)**:
You don't need to run the perfscale-audit tool, just check prometheus.
1. Deploying Kubevirt and immediately running the test that creates and deletes a VMI
** I deployed Kubevirt using `make cluster-sync`, but you can also use `make cluster-deploy`
** Just create and delete a VMI, or run `make perftest`
2. Have Kubevirt already deployed to the system for a while and then run the test. Although CREATE event appears in the metric, the DELETE event takes a while to appear, I have to wait 3 minutes after finishing the test for all events appear.

**Anything else we need to know?**:
I will describe 3 different scenarios to illustrate the problem.

**Scenario 1: Creating 100 VMI just after deploying kubevirt**
kubevirt deployed: 12:52:58
test started: 12:53:01
test stopped: 12:55:16
image
There are no pod CREATE and DELETE events appearing in the query.

**Scenario 2: Creating 100 VMI just after running the previous test kubevirt (Have Kubevirt already deployed)**
kubevirt deployed: 12:52:58
test started: 13:01:00
test stopped: 13:03:33
image
Both CREATE and DELETE events are appearing in the query.
Also, note that there are many other events appearing in the query than in the previous scenario.
Another interesting factor is that the pod's first DELETE event appeared at 13:04:56 and the last at 13:09:25, long after the test ended at 13:03:33.

**Scenario 3: Deploy Kubevirt, create 1 VMI, wait 30s and create 100 VMIs**
image
kubevirt deployed: 13:21:10
test started: 13:22:39
test stopped: 13:25:00
I'm waiting 30s after "warm up" to let Prometheus scrape the metrics before running the second test.
The pod's first DELETE event appeared at 13:26:29 and the last at 12:31:27, long after the test ended at 13:25:00, as in the previous scenario.

**Workaround**
If I "warm up" the system, such as creating and deleting a VMI, the expected events show up in the prometheus metric.
Also, as a workaround to collect the DELETE metrics from the pod, I need to wait a few minutes after finishing the test to collect the Prometheus metrics.

**Final remarks**
The problem does not seem to be related to waiting for the system to initialize. I tried deploying kubevirt, waiting 15m and then running the test, but it was still missing events. The experiment only behaves as expected when I "warm up" the system.

Although I'm not sure what the root cause is, it only happens when the test is run right after installing Kubevirt. 
To resolve the issue, we need further investigation of the code, verifying which component is responsible for reporting the rest client requests related to pod events.

**Environment**:
- KubeVirt version (use `virtctl version`):
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by marceloamaral**
**What this PR does / why we need it**:
Problem: [Performance job is failing](https://testgrid.k8s.io/kubevirt-periodics#periodic-kubevirt-performance-cluster-100-density-test&width=20), VMIs are not being created.

Previous PR (#7265) updated the perfscale-load-generator to introduce a new test type: the steady-state test.
However, this previous PR removed a lot of configurations and reordered some parts of the code that shouldn't be changed.
For example, a watcher must be started before creating objects to be able to watch all created objects. And, as in the previous version, we need to create the namespace that the test will run on.

In this PR, I fix those problems and also reverted the test UUID to be per test and not per work. Because if the create job creates objects with a label with their UUID, the delete job cannot find the objects that the other job created if the UUIDs are different.
Also, as in the previous version, I added again the option to run a test that creates and also deletes the objects. The current version has these two mutually exclusive.

This PR also fixes the steady-state test, introducing ramp-up, steady-state, and ramp-down phases (the importance of this is [here](https://www.loadview-testing.com/blog/the-importance-of-ramp-up-and-ramp-down-user-load/)).
Also, the steady state wait function has been updated, where the wait will be either minChurnSleep or DefaultMinSleepChurn.
Moreover, in steady state testing, the delete function was not deleting just a subset of the object, but was deleting all objects in the namespaces. So doing something different than what the churn parameter expected. I.e., recreating X objs.

Finally, this PR fixes a previous metric collection issue (#7083). In kubevirt CI/CD, the node where we run the job is on a different cluster, so time is not synchronized across all nodes. So the timestamps collected from the test are different in Prometheus, so the test doesn't collect metrics correctly. To fix it, we collect the timestamps by doing a prometheus query so that we can use the exact time that prometheus is using.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7083

**Special notes for your reviewer**:
I tested both burst and steady state tests in our CI/CD system.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by marceloamaral** [comment]
/cc @rthallisey @dhiller 

As mentioned before, the performance jobs are not running for a long time, since the tool was updated. We need to fix it as soon as possible.
So, could you please give priority to this PR?

File: tools/perfscale-load-generator/flags/flags.go [hunk 1]
```diff
@@ -42,6 +43,7 @@ func init() {
 	flag.IntVar(&Verbosity, "v", 2, "log level for V logs")
 	flag.StringVar(&WorkloadConfigFile, "workload", "tools/perfscale-load-generator/examples/workload/kubevirt-density/kubevirt-density.yaml", "path to the file containing the worload configuration")
 	flag.BoolVar(&Delete, "delete", false, "Delete a workload")
+	flag.BoolVar(&Delete, "delete-only", false, "Delete a workload")
```
Comment (VirrageS) [inline tools/perfscale-load-generator/flags/flags.go:46]:
```suggestion
	flag.BoolVar(&DeleteOnly, "delete-only", false, "Delete a workload")
```

File: tools/perfscale-load-generator/load-generator.go [hunk 2]
```diff
@@ -45,21 +47,24 @@ func main() {
 		workload.Timeout.Duration = config.Timeout
 	}
 
+	testUUID, _ := uuid.NewUUID()
```
Comment (VirrageS) [inline tools/perfscale-load-generator/load-generator.go:50]:
```suggestion
	testUUID := uuid.New().String()
```

File: tools/perfscale-load-generator/object/namespace.go [hunk 3]
```diff
@@ -32,7 +32,12 @@ import (
 	"kubevirt.io/client-go/log"
 )
 
-func CreateNamespace(virtCli kubecli.KubevirtClient, name, scenarioLabel, uuid string) error {
+var createdNamespaceCache = map[string]bool{}
```
Comment (VirrageS) [inline tools/perfscale-load-generator/object/namespace.go:35]: Is it really needed? It feels like some optimization that can backfire easily - the cache is not updated for the namespace that were deleted, so we might not be creating namespaces that should be created.

File: tools/perfscale-load-generator/steady-state/steady-state.go [hunk 4]
```diff
@@ -36,169 +34,170 @@ import (
...
 
 // NewSteadyStateJob
-func newSteadyStateJob(virtClient kubecli.KubevirtClient, workload *config.Workload) *SteadyStateJob {
-	uid, _ := uuid.NewUUID()
+func newSteadyStateJob(virtClient kubecli.KubevirtClient, workload *config.Workload, uuid string) *SteadyStateJob {
+	// minChurnSleep is an optinal config
+	if workload.MinChurnSleep == nil {
+		workload.MinChurnSleep = &config.Duration{Duration: config.DefaultMinSleepChurn}
+	}
+	if workload.MinChurnSleep.Duration < time.Duration(1*time.Microsecond) {
```
Comment (VirrageS) [inline tools/perfscale-load-generator/steady-state/steady-state.go:59]:
```suggestion
	if workload.MinChurnSleep.Duration < time.Microsecond {
```

File: tools/perfscale-load-generator/steady-state/steady-state.go [hunk 5]
```diff
@@ -36,169 +34,170 @@ import (
...
-	log.Log.V(1).Infof("SteadyState Load Generator Wait")
-	if b.minChurnSleep.Duration < time.Duration(1*time.Microsecond) {
-		time.Sleep(config.DefaultMinSleepChurn)
-		return
+	timePassed := time.Since(b.currTimestamp)
+	remainingTime := b.Workload.Timeout.Duration - timePassed
+	log.Log.V(3).Infof("Wait %f seconds before recreating %d objs. Remaining %f seconds to finish the test", b.minChurnSleep.Seconds(), b.Workload.Churn, remainingTime.Seconds())
+	time.Sleep(b.minChurnSleep.Duration)
+	// time.Sleep(remainingTime)
+	// b.currTimestamp = time.Now()
```
Comment (VirrageS) [inline tools/perfscale-load-generator/steady-state/steady-state.go:184]: Is this commented correctly? If yes, then maybe it should be removed and the `currTimestamp` should be renamed to `startedTime`?

File: tools/perfscale-load-generator/watcher/watcher.go [hunk 6]
```diff
@@ -122,6 +124,9 @@ func (w *ObjListWatcher) handleUpdate(obj interface{}) {
 		if vmi.Status.Phase == kvv1.Running {
 			w.addObj(string(vmi.GetUID()))
 		}
+		if vmi.Status.Phase == kvv1.Succeeded {
```
Comment (VirrageS) [inline tools/perfscale-load-generator/watcher/watcher.go:127]: What about `kvv1.Failed`? Shouldn't we also `w.deleteObj`?

File: tools/perfscale-load-generator/config/types.go [hunk 7]
```diff
@@ -34,14 +35,25 @@ const (
...
+var (
 	ContainerPrefix = "registry:5000/kubevirt"
 	ContainerTag    = "devel"
-	Timeout         = time.Duration(5 * time.Minute)
 )
 
-//type Duration time.Duration
+func init() {
+	if os.Getenv("DOCKER_PREFIX") != "" {
+		ContainerPrefix = os.Getenv("DOCKER_PREFIX")
```
Comment (VirrageS) [inline tools/perfscale-load-generator/config/types.go:51]:
```suggestion
	if dockerPrefixEnv := os.Getenv("DOCKER_PREFIX"); dockerPrefixEnv != "" {
		ContainerPrefix = dockerPrefixEnv
```

File: tools/perfscale-load-generator/config/types.go [hunk 8]
```diff
@@ -34,14 +35,25 @@ const (
...
+	}
+	if os.Getenv("DOCKER_TAG") != "" {
+		ContainerTag = os.Getenv("DOCKER_TAG")
```
Comment (VirrageS) [inline tools/perfscale-load-generator/config/types.go:54]:
```suggestion
	if dockerTagEnv := os.Getenv("DOCKER_TAG"); dockerTagEnv != "" {
		ContainerTag = dockerTagEnv
```

**Review by VirrageS — CHANGES_REQUESTED** [review]

**Comment by dhiller** [comment]
> @VirrageS: changing LGTM is restricted to collaborators

I figure this is related to you not (yet) being a KubeVirt community member. Please see [member due diligence](https://github.com/kubevirt/community/tree/main/member_dd) for how to apply.

**Comment by dhiller** [comment]
@VirrageS ^^

File: hack/cluster-clean.sh [hunk 9]
```diff
@@ -26,7 +26,7 @@ source cluster-up/cluster/$KUBEVIRT_PROVIDER/provider.sh
 source hack/config.sh
 
 function patch_remove_finalizers() {
-    _kubectl patch --type=json -p '[{ "op": "remove", "path": "/metadata/finalizers" }]' $@
+    _kubectl patch --type=json -p '[{ "op": "remove", "path": "/metadata/finalizers" }]' $@ || true
```
Comment (dhiller) [inline hack/cluster-clean.sh:29]: This looks like it would mask errors. Why is this necessary?

File: hack/perfscale-tests.sh [hunk 10]
```diff
@@ -63,22 +71,24 @@ if [[ ${PERFAUDIT} == "true" || ${PERFAUDIT} == "True" ]]; then
...
-    stop_timestamp=$(date -u +%Y-%m-%dT%TZ)
+    # wait 30s after finished the test. More info in https://github.com/kubevirt/kubevirt/issues/7083
+    echo "Sleeping 30s to let prometheus scrape all metrics"
+    sleep 30s
+    stop_timestamp=$(get_timestamp)
     perfaudit $start_timestamp $stop_timestamp
 fi
+
+echo "start_timestamp= $start_timestamp"
+echo "stop_timestamp= $stop_timestamp"
```
Comment (dhiller) [inline hack/perfscale-tests.sh:94]: I don't see `stop_timestamp` being set besides the above `if`? Does it matter?

File: tools/perfscale-load-generator/burst/burst.go [hunk 11]
```diff
@@ -44,104 +44,115 @@ type BurstJob struct {
...
+	// This is important because before creating each obj we need to create a obj watcher for each obj type.
+	objSpec := b.Workload.Object
+	objSample := renderObjSpecTemplate(objSpec, b.UUID)
+	b.createWatcherIfNotExist(objSample)
+
+	// Create all replicas
+	for r := 1; r <= b.Workload.Count; r++ {
+		log.Log.V(2).Infof("Replica %d of %d", r, b.Workload.Count)
+		idx := r
+		_, err := utils.CreateObjectReplica(b.virtClient, objSpec, &idx, b.UUID)
```
Comment (dhiller) [inline tools/perfscale-load-generator/burst/burst.go:89]: nit: this looks like a lack of cohesion, thus putting it into `utils`? Maybe there's a better place for this function?

File: tools/perfscale-load-generator/burst/burst.go [hunk 12]
```diff
@@ -44,104 +44,115 @@ type BurstJob struct {
...
-		wg.Add(1)
-		go func(newObject *unstructured.Unstructured) {
-			defer wg.Done()
-			b.watchCreate(newObject)
-			log.Log.Infof("obj %s is available", newObject.GroupVersionKind().Kind)
-		}(newObject)
+	// Wait all objects be Running
+	for objType, objWatcher := range b.watchers {
+		log.Log.Infof("Wait for obj(s) %s to be available", objType)
+		objWatcher.WaitRunning(b.Workload.Timeout.Duration)
```
Comment (dhiller) [inline tools/perfscale-load-generator/burst/burst.go:98]: nit: shouldn't the timeout be a property of the watcher?

File: tools/perfscale-load-generator/burst/burst.go [hunk 13]
```diff
@@ -44,104 +44,115 @@ type BurstJob struct {
...
+	objType := objUtil.GetObjectResource(objSample)
+	labels := objSample.GetLabels()
+	jobUUID := labels[config.WorkloadLabel]
+	log.Log.V(2).Infof("Deleting %d objects for job %s", b.Workload.Count, jobUUID)
+	objUtil.DeleteNObjectsInNamespaces(b.virtClient, objType, config.GetListOpts(config.WorkloadLabel, jobUUID), b.Workload.Count)
+
+	// Wait all objects be Deleted. In the case of VMI, deleted means the succeded phase.
+	for objType, objWatcher := range b.watchers {
+		log.Log.Infof("Wait for obj(s) %s to be deleted", objType)
+		objWatcher.WaitDeletion(b.Workload.Timeout.Duration)
```
Comment (dhiller) [inline tools/perfscale-load-generator/burst/burst.go:118]: nit: shouldn't the timeout be a property of the watcher?

File: tools/perfscale-load-generator/burst/burst.go [hunk 14]
```diff
@@ -44,104 +44,115 @@ type BurstJob struct {
...
-		objUtil.DeleteAllObjectsInNamespaces(b.virtClient, objType, config.GetListOpts(config.WorkloadLabel, jobUUID))
-		b.watchDelete(getObject)
+	objSpec := b.Workload.Object
+	objSample := renderObjSpecTemplate(objSpec, b.UUID)
+	b.createWatcherIfNotExist(objSample)
+
+	// Delete objects that match labels up to the count
+	objType := objUtil.GetObjectResource(objSample)
+	labels := objSample.GetLabels()
+	jobUUID := labels[config.WorkloadLabel]
```
Comment (dhiller) [inline tools/perfscale-load-generator/burst/burst.go:111]: nit: maybe rename the config property to reflect it's value?

File: tools/perfscale-load-generator/burst/burst.go [hunk 15]
```diff
@@ -44,104 +44,115 @@ type BurstJob struct {
...
-		b.Workload.Count,
-		*config.GetListOpts(config.WorkloadLabel, b.UUID))
-	objWatcher.Run()
-
-	log.Log.Infof("Wait for obj(s) %s to be available", b.objType)
-	objWatcher.WaitRunning(b.Workload.Timeout.Duration)
-	objWatcher.Stop()
+func (b *BurstJob) createWatcherIfNotExist(obj *unstructured.Unstructured) {
+	objType := objUtil.GetObjectResource(obj)
+	if _, exist := b.watchers[objType]; !exist {
```
Comment (dhiller) [inline tools/perfscale-load-generator/burst/burst.go:126]: nit: this is not thread safe, but we do not need to guard for parallelization here, right?

File: tools/perfscale-load-generator/load-generator.go [hunk 16]
```diff
@@ -45,21 +47,24 @@ func main() {
...
+		lg = &burst.BurstLoadGenerator{Done: timeout, UUID: testUUID.String()}
 	} else if workload.Type == "steady-state" {
-		lg = &steadyState.SteadyStateLoadGenerator{Done: timeout}
+		lg = &steadyState.SteadyStateLoadGenerator{Done: timeout, UUID: testUUID.String()}
 	} else {
 		log.Log.V(1).Errorf("Load Generator doesn't have type %s", workload.Type)
 		return
 	}
 
+	if !flags.DeleteOnly {
```
Comment (dhiller) [inline tools/perfscale-load-generator/load-generator.go:63]: nit: maybe rename property to `Run`?

File: tools/perfscale-load-generator/object/unstructured.go [hunk 17]
```diff
@@ -65,49 +64,27 @@ func ListObjects(virtCli kubecli.KubevirtClient, resourceKind string, listOpts *
...
-	}
-	log.Log.V(2).Infof("Didn't find any matching objects. The previous job had already been cleaned up")
-	return nil, ""
-}
-
-// DeleteAllObjectsInNamespaces deletes a collection of objects in a set of namespace with a given selector
-func DeleteAllObjectsInNamespaces(virtCli kubecli.KubevirtClient, resourceKind string, listOpts *metav1.ListOptions) {
+// DeleteNObjectsInNamespaces deletes n objects in a set of namespace with a given selector.
+// This is used to create churn in the steady state test
+func DeleteNObjectsInNamespaces(virtCli kubecli.KubevirtClient, resourceKind string, listOpts *metav1.ListOptions, count int) {
```
Comment (dhiller) [inline tools/perfscale-load-generator/object/unstructured.go:75]: nit: rename parameter to make usage more clear
```suggestion
func DeleteNObjectsInNamespaces(virtCli kubecli.KubevirtClient, resourceKind string, listOpts *metav1.ListOptions, n int) {
```

File: tools/perfscale-load-generator/steady-state/steady-state.go [hunk 18]
```diff
@@ -36,169 +34,170 @@ import (
...
+	ss.stopAllWatchers()
 	return
 }
 
 func (b *SteadyStateLoadGenerator) Run(virtClient kubecli.KubevirtClient, workload *config.Workload) {
-	ss := newSteadyStateJob(virtClient, workload)
+	ss := newSteadyStateJob(virtClient, workload, b.UUID)
+	// Ramp up phase, creating all objects, waiting before starting the steady state phase
+	log.Log.V(1).Infof("Starting the Ramp Up Phase")
+	ss.CreateWorkload(ss.Workload.Count)
```
Comment (dhiller) [inline tools/perfscale-load-generator/steady-state/steady-state.go:87]: nit: plural?
```suggestion
	ss.CreateWorkloads(ss.Workload.Count)
```

File: tools/perfscale-load-generator/steady-state/steady-state.go [hunk 19]
```diff
@@ -36,169 +34,170 @@ import (
...
-	if getObject != nil {
-		labels := getObject.GetLabels()
-		jobUUID := labels[config.WorkloadLabel]
-		log.Log.V(2).Infof("Deleting all workloads for job %s", jobUUID)
-		objUtil.DeleteAllObjectsInNamespaces(virtClient, objType, config.GetListOpts(config.WorkloadLabel, jobUUID))
-		ss.watchDelete(getObject)
-	}
-	log.Log.V(2).Infof("All workloads for job have been deleted")
+	ss := newSteadyStateJob(virtClient, workload, b.UUID)
+	ss.DeleteWorkload(ss.Workload.Count)
```
Comment (dhiller) [inline tools/perfscale-load-generator/steady-state/steady-state.go:77]:
```suggestion
	ss.DeleteWorkloads(ss.Workload.Count)
```

File: tools/perfscale-load-generator/steady-state/steady-state.go (hunk 5 above)
Comment (marceloamaral) [reply tools/perfscale-load-generator/steady-state/steady-state.go:184]: It's a leftover, I'll update it

File: tools/perfscale-load-generator/watcher/watcher.go (hunk 6 above)
Comment (marceloamaral) [reply tools/perfscale-load-generator/watcher/watcher.go:127]: Yes, it should. Thanks!

File: hack/cluster-clean.sh (hunk 9 above)
Comment (marceloamaral) [reply hack/cluster-clean.sh:29]: We don't need anymore..

`make cluster-clean` was deleting the created VMIs and fixing them to remove them. However, during patching some VMs were actually deleted and then the patch command failed as the obj no longer existed....

Anyway, I will remove this

File: tools/perfscale-load-generator/watcher/watcher.go (hunk 6 above)
Comment (marceloamaral) [reply tools/perfscale-load-generator/watcher/watcher.go:127]: In fact, when the VM enters the "Failed" phase, it will be moved to the "Succeeded" phase later. Therefore, we must wait for the `Succeeded` phase to delete the obj entry.

File: hack/perfscale-tests.sh (hunk 10 above)
Comment (marceloamaral) [reply hack/perfscale-tests.sh:94]: Will fix it. Thanks

File: tools/perfscale-load-generator/burst/burst.go (hunk 11 above)
Comment (marceloamaral) [reply tools/perfscale-load-generator/burst/burst.go:89]: Yes, it makes sense. Will move it to the object package

File: tools/perfscale-load-generator/burst/burst.go (hunk 12 above)
Comment (marceloamaral) [reply tools/perfscale-load-generator/burst/burst.go:98]: Maybe because right now we have only one kind of timeout. But we could have different ones based on the obj type and test configuration....

File: tools/perfscale-load-generator/burst/burst.go (hunk 15 above)
Comment (marceloamaral) [reply tools/perfscale-load-generator/burst/burst.go:126]: We do not run parallel tests, so there is only one thread accessing the `b.watchers[objType]`

**Comment by marceloamaral** [comment]
@dhiller and @VirrageS I updated the code considering all your comments.
It is ready for the next round of review.

**Review by rthallisey — CHANGES_REQUESTED** [review]

**Comment by maiqueb** [comment]
/uncc

No clue about this.

File: tools/perfscale-load-generator/burst/burst.go [hunk 20]
```diff
@@ -44,104 +43,116 @@ type BurstJob struct {
...
+	}
 
-		wg.Add(1)
-		go func(newObject *unstructured.Unstructured) {
-			defer wg.Done()
-			b.watchCreate(newObject)
-			log.Log.Infof("obj %s is available", newObject.GroupVersionKind().Kind)
-		}(newObject)
+	// Wait all objects be Running
+	for objType, objWatcher := range b.watchers {
```
[the remark this answers is not in the document]
Comment (dhiller) [reply tools/perfscale-load-generator/burst/burst.go:96]: I think the assumption is that all VMs should be in running state anyway, thus I think the order is irrelevant here. Am I wrong about this? Or do you mean that there might be a kind of wait starvation, in the kind of, "lots of other VMs are coming up before the one I am waiting for currently"?

[the remark this answers is not in the document]
Comment (rthallisey) [reply tools/perfscale-load-generator/burst/burst.go:96]: I thought `objWatcher.WaitRunning(b.Workload.Timeout.Duration)` is being called for each object.  My concern is that we're waiting for one object to be completed before we start the timeout for the next one.  I might be getting something confused... because why the for loop...?

[the remark this answers is not in the document]
Comment (marceloamaral) [reply tools/perfscale-load-generator/burst/burst.go:96]: Actually we have a watcher for each type of object. So the loop exist to wait all obj for each types.

Having the logic to create a watcher for each obj was one of the main thing that I fixed. A watcher will watch all obj for a given type...

In the previous version, it was possible to create different types of object at the same time, for example mix Pods with VMs, or VM and VMI, etc....

File: tools/perfscale-load-generator/burst/burst.go [hunk 21]
```diff
@@ -44,104 +43,116 @@ type BurstJob struct {
...
+		UUID:       uuid,
 		done:       d,
+		watchers:   map[string]*watcher.ObjListWatcher{},
 	}
 }
 
 func (b *BurstLoadGenerator) Delete(virtClient kubecli.KubevirtClient, workload *config.Workload) {
-	j := newBurstJob(virtClient, workload, b.Done)
-	j.DeleteWorkload()
+	j := newBurstJob(virtClient, workload, b.UUID, b.Done)
```
[the remark this answers is not in the document]
Comment (marceloamaral) [reply tools/perfscale-load-generator/burst/burst.go:61]: I think that the idea of this code was enable BurstLoadGenerator to create many jobs. In this case, we cannot merge them.

File: tools/perfscale-load-generator/burst/burst.go [hunk 22]
```diff
@@ -44,104 +43,116 @@ type BurstJob struct {
...
+	objType := objUtil.GetObjectResource(objSample)
+	labels := objSample.GetLabels()
+	jobUUID := labels[config.WorkloadUUIDLabel]
+	log.Log.V(2).Infof("Deleting %d objects for job %s", b.Workload.Count, jobUUID)
+	objUtil.DeleteNObjectsInNamespaces(b.virtClient, objType, config.GetListOpts(config.WorkloadUUIDLabel, jobUUID), b.Workload.Count)
+
+	// Wait all objects be Deleted. In the case of VMI, deleted means the succeded phase.
+	for objType, objWatcher := range b.watchers {
+		log.Log.Infof("Wait for obj(s) %s to be deleted", objType)
+		objWatcher.WaitDeletion(b.Workload.Timeout.Duration)
```
[the remark this answers is not in the document]
Comment (marceloamaral) [reply tools/perfscale-load-generator/burst/burst.go:118]: I think we can change this later in another PR.

**Comment by marceloamaral** [comment]
@rthallisey could you please review again and see if you agree with my comments?

**Comment by rthallisey** [comment]
/lgtm

We'll discuss any refactoring in SIG-Scale and address in a follow up.

**Comment by marceloamaral** [comment]
@dhiller could you approve this PR?

**Comment by dhiller** [comment]
/approve

Sorry for the delay.
